### PR TITLE
Psp 3646 warn user when navigating away from property details.

### PR DESCRIPTION
**Pull request by devinleighsmith**

### CODE REVIEW

File: frontend/src/components/maps/leaflet/mapUtils.tsx [hunk 1]
```diff
@@ -161,7 +161,7 @@ export const toCqlFilter = (object: any, pidOverride?: boolean) => {
   const cql: string[] = [];
   Object.keys(object).forEach((key: string) => {
     if (object[key]) {
-      if (key === 'PID' && object[key] && !pidOverride) {
+      if (key === 'PID' || (key === 'PID_PADDED' && object[key] && !pidOverride)) {
```
Comment (asanchezr) [inline frontend/src/components/maps/leaflet/mapUtils.tsx:164]: Question - is this change related to this PR? I saw another PR with this change already...

File: frontend/src/features/mapSideBar/tabs/propertyDetails/update/UpdatePropertyDetailsForm.tsx [hunk 2]
```diff
@@ -86,6 +87,7 @@ export const UpdatePropertyDetailsForm: React.FC<IUpdatePropertyDetailsFormProps
 
   return (
     <StyledForm>
+      <UnsavedChangesPrompt />
```
Comment (asanchezr) [inline frontend/src/features/mapSideBar/tabs/propertyDetails/update/UpdatePropertyDetailsForm.tsx:90]: This will prevent navigating out of the form if there are unsaved changes but don't we want to also confirm upon cancel button click?

File: frontend/src/features/properties/selector/search/PropertySelectorSearchContainer.test.tsx [hunk 3]
```diff
@@ -61,7 +61,7 @@ describe('PropertySelectorSearchContainer component', () => {
     userEvent.click(searchButton);
     await waitFor(() => {
       expect(mockAxios.history.get[0].url).toBe(
-        "https://openmaps.gov.bc.ca/geo/pub/WHSE_CADASTRE.PMBC_PARCEL_FABRIC_POLY_SVW/wfs?service=WFS&REQUEST=GetFeature&VERSION=1.3.0&outputFormat=application/json&typeNames=pub:WHSE_CADASTRE.PMBC_PARCEL_FABRIC_POLY_SVW&srsName=EPSG:4326&cql_filter=PID ilike '%25123456789%25'",
+        "https://openmaps.gov.bc.ca/geo/pub/WHSE_CADASTRE.PMBC_PARCEL_FABRIC_POLY_SVW/wfs?service=WFS&REQUEST=GetFeature&VERSION=1.3.0&outputFormat=application/json&typeNames=pub:WHSE_CADASTRE.PMBC_PARCEL_FABRIC_POLY_SVW&srsName=EPSG:4326&cql_filter=PIN ilike '%25123456789%25' OR PID ilike '%25123456789%25'",
```
Comment (asanchezr) [inline frontend/src/features/properties/selector/search/PropertySelectorSearchContainer.test.tsx:64]: Seems unrelated to this PR

File: frontend/src/features/mapSideBar/tabs/propertyDetails/update/UpdatePropertyDetailsForm.tsx (hunk 2 above)
Comment (devinleighsmith) [reply frontend/src/features/mapSideBar/tabs/propertyDetails/update/UpdatePropertyDetailsForm.tsx:90]: when clicking the cancel button the action attempts to change the route, which also triggers this.
